### DISCUSSION TITLE
Is there a way to prevent re-render with <Match keyed> ?

### DISCUSSION THREAD

**Discussion by yw662** [solidjs/solid#1352] [Q&A] [answered]
Exp in this
```ts
const [ resource, { refetch } ] = createResource(...)

return <Switch fallback={...}>
<Match when={resource.latest}>
<Foo bar={resource.latest!} />
</Match>
</Switch>
```

with keyed Match, the Foo line can be `latest => <Foo bar={latest} />`, and the assertion can be omitted.
However, if `refetch` is called, keyed Match will need to fully re-render `Foo`, while non-keyed Match does not.

Is there a way to combine the good parts, so that there will be no assertion, and no re-renders ?

**Comment by yw662** [comment]
Of course `eslint-disable-next-line @typescript-eslint/no-non-null-assertion` is an option since we indeed outsmart typescript and eslint on this particular case. However is there a way to comfort them ?

**Comment by ryansolid** [comment, marked as answer]
This is what keyed is and it has its use cases unfortunately we shipped that as the default for callbacks with 1.0 and retroactively added keyed as a way to inform people this would be the case and open up the potential for future breaking changes where the callback without `keyed` would work by passing a narrowed accessor through. Unfortunately the API as of today can do this without a breaking change which you can expect in Solid 2.0.
